File: web/src/lib/affiliates.test.ts
import { describe, it, expect } from "vitest";
import { parseAffiliates } from "./affiliates";

describe("parseAffiliates", () => {
  const raw = {
    teams: [
      { id: 1, name: "MLB Club", sport: { id: 1 }, league: { name: "NL" } },
      { id: 2, name: "AAA Club", sport: { id: 11 }, league: { name: "IL" } },
      { id: 3, name: "Rookie Club", sport: { id: 16 }, league: { name: "ACL" } },
      { id: 4, name: "AA Club", sport: { id: 12 }, league: { name: "TL" } },
      { id: 5, name: "Org Pseudo", sport: { id: 21 }, league: { name: "MiLB" } },
    ],
  };

  it("filters to real levels, labels them, and orders AAA→Rookie", () => {
    const a = parseAffiliates(raw);
    expect(a.map((x) => x.level)).toEqual(["AAA", "AA", "Rookie"]);
    expect(a.map((x) => x.id)).toEqual([2, 4, 3]);
    expect(a[0].league).toBe("IL");
  });

  it("returns [] for empty/garbage", () => {
    expect(parseAffiliates({})).toEqual([]);
    expect(parseAffiliates({ teams: [] })).toEqual([]);
  });
});

File: cli/internal/mlb/affiliates_test.go
package mlb

import "testing"

func TestFilterAffiliates(t *testing.T) {
	mk := func(id, sportID int, name string) Team {
		var tm Team
		tm.ID = id
		tm.Name = name
		tm.Sport.ID = sportID
		return tm
	}
	in := []Team{
		mk(1, 1, "MLB Club"),
		mk(2, 16, "Rookie Club"),
		mk(3, 11, "AAA Club"),
		mk(4, 21, "Org Pseudo"),
		mk(5, 12, "AA Club"),
	}
	out := FilterAffiliates(in)
	var got []string
	for _, x := range out {
		got = append(got, x.Name)
	}
	want := []string{"AAA Club", "AA Club", "Rookie Club"}
	if len(got) != len(want) {
		t.Fatalf("got %v, want %v", got, want)
	}
	for i := range want {
		if got[i] != want[i] {
			t.Fatalf("order: got %v, want %v", got, want)
		}
	}
}
